components/org.wso2.carbon.identity.application.authentication.handler.identifier/src/main/java/org/wso2/carbon/identity/application/authentication/handler/identifier/IdentifierHandler.java

```diff
@@ -332,7 +334,17 @@ protected void processAuthenticationResponse(HttpServletRequest request,
         FrameworkUtils.validateUsername(request.getParameter(BasicAuthenticatorConstants.USER_NAME), context);
         String username = FrameworkUtils.preprocessUsername(
                 request.getParameter(IdentifierHandlerConstants.USER_NAME), context);
-
+        if (IdentifierAuthenticatorServiceComponent.getMultiAttributeLogin().isEnabled(context.getTenantDomain())) {
+            ResolvedUserResult resolvedUserResult = IdentifierAuthenticatorServiceComponent.getMultiAttributeLogin().
+                    resolveUser(MultitenantUtils.getTenantAwareUsername(username), context.getTenantDomain());
+            if (resolvedUserResult != null && ResolvedUserResult.UserResolvedStatus.SUCCESS.
+                    equals(resolvedUserResult.getResolvedStatus())) {
+                username = UserCoreUtil.addTenantDomainToEntry(resolvedUserResult.getUser().getUsername(),
+                        context.getTenantDomain());
+            } else {
+                throw new InvalidCredentialsException("User not exist");
+            }
+        }
```

Comment:
what about moving this logic inside FrameworkUtils.preprocessUsername(). So we can reduce code duplication and maintain easily

Reply:
Yes we can move this in to FrameworkUtils.  But not in to preprocessUsername(), because several other methods(which are use multi attribute login in other repo ) does not use preprocessUsername(). So we can move this logic in to FrameworkUtils as a separate method.

Reply:
fixed f5dc1b8